### PR TITLE
chore: support admin key overrides

### PR DESCRIPTION
**Pull request by tinker-michaelj**
**Description**:
  - Cherry-pick #16781

### CODE REVIEW

File: hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java [hunk 1]
```diff
@@ -150,76 +157,93 @@ public void doPostUpgradeSetup(@NonNull final Dispatch dispatch) {
...
+        final var autoNodeAdminKeyUpdates = new AutoEntityUpdate<Map<Long, Key>>(
+                (ctx, nodeAdminKeys) -> nodeAdminKeys.forEach(
+                        (nodeId, key) -> ctx.dispatchAdmin(b -> b.nodeUpdate(NodeUpdateTransactionBody.newBuilder()
+                                .nodeId(nodeId)
+                                .adminKey(key)
+                                .build()))),
+                adminConfig.upgradeNodeAdminKeysFile(),
+                SystemSetup::parseNodeAdminKeys);
+        if (autoNodeAdminKeyUpdates.tryIfPresent(adminConfig.upgradeSysFilesLoc(), systemContext)) {
+            dispatch.stack().commitFullStack();
```
Comment (iwsimon) [inline hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java:195]: I noticed `fileService.updateAddressBookAndNodeDetailsAfterFreeze(systemContext, nodeStore);` was not moved after this in release/0.57 and here. We updated it in release/0.56 branch.

Comment (tinker-michaelj) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java:195]: Hi @iwsimon, wasn't that only a concern in 0.56 when we had [this](https://github.com/hashgraph/hedera-services/blob/release/0.56/hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java#L195) block to initialize node metadata from the pre-DAB state?

Comment (iwsimon) [reply hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/record/SystemSetup.java:195]: Yea, you are right. NodeMetadataHelper is not used any more.

**Review by iwsimon — APPROVED** [review]
LGTM, thanks @tinker-michaelj

**Review by anthony-swirldslabs — APPROVED** [review]
* `platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/config/internal/PlatformConfigUtils.java`

LGTM

**Pull request merged**